packages/product/src/repositories/product-tag.ts

```diff
@@ -91,32 +94,43 @@ export class ProductTagRepository extends AbstractBaseRepository<ProductTag> {
       if (aTag) {
         upsertedTags.push(aTag)
       } else {
-        const newTag = manager.create(ProductTag, tag)
+        const newTag = (manager as SqlEntityManager).create(ProductTag, tag)
         tagsToCreate.push(newTag)
       }
     })
 
     if (tagsToCreate.length) {
       const newTags: ProductTag[] = []
       tagsToCreate.forEach((tag) => {
-        newTags.push(manager.create(ProductTag, tag))
+        newTags.push((manager as SqlEntityManager).create(ProductTag, tag))
       })
 
-      await manager.persist(newTags)
+      await (manager as SqlEntityManager).persist(newTags)
       upsertedTags.push(...newTags)
     }
 
     return upsertedTags
   }
 
-  async delete(ids: string[], context: Context = {}): Promise<void> {
-    const manager = (context.transactionManager ??
-      this.manager_) as SqlEntityManager
-
-    await manager.nativeDelete(Product, { id: { $in: ids } }, {})
+  @InjectEntityManager(doNotForceTransaction, "")
```

Comment:
I don't like the empty string here. Can we make the argument optional for these cases?

Reply:
the argument is already optional and have a default value being manager_

Reply:
that it should use `this.transaction` and not `this.something.transaction`

Reply:
and passing false would means that there is not a property to find on this and use this directly instead

Reply:
I would create a new parameter `useThis` if the argument was an object. but if I move to an object then I need to refactor all the existing modules. wdyt?

Reply:
Let's keep it as is 👍